src/osbuild-manifests/aws_tag.py

```diff
@@ -1,79 +1,95 @@
-            print(f"The build is {build_id}")
+            print(f"The build is {build_id} for {arch}")
             buildFetch(args.stream, build_id, arch)
-            meta = open('builds/'+build_id+'/'+arch+'/meta.json')
+            meta = open(f'builds/{build_id}/{arch}/meta.json')
             data = json.load(meta)
-
+ 
             # Delete this when actually running. Just here while I make this script
-            data ={"amis":[{
-                "name": "us-east-1",
-                "hvm": "ami-0016d5df3041499f9",
-                "snapshot": "snap-0c1ca4850fcd5e573"
-            }]}
-            amis = data['amis']
+            # data ={"amis":[{
+            #     "name": "us-east-1",
+            #     "hvm": "ami-0016d5df3041499f9",
+            #     "snapshot": "snap-0c1ca4850fcd5e573"
+            # }]}
+            if 'amis' in data.keys():
+                amis = data['amis']
+            else:
+                print(f"{build_id} does not have any AMIs for {arch} in meta.json")
+                return
             for ami in amis:
-                checkAndAddTag(ami["hvm"], ami["name"])
-                checkAndAddTag(ami["snapshot"], ami["name"])
-            return
-
+                if args.dry_run:
+                    amiTagExists = checkTag(ami["hvm"], ami["name"])
+                    snapshotTagExists = checkTag(ami["snapshot"], ami["name"])
+                    if not amiTagExists:
+                        print(f"Would add tags 'FedoraGroup=coreos' to {ami["hvm"]} in region {ami["name"]}")
+                    else:
+                       print(f"{ami["hvm"]} already tagged with FedoraGroup=coreos tag")
+                    if not snapshotTagExists:
+                        print(f"Would add tags 'FedoraGroup=coreos' to {ami["snapshot"]} in region {ami["name"]}")
+                    else:
+                       print(f"{ami["snapshot"]} already tagged with FedoraGroup=coreos tag")
+                else:
+                    checkAndAddTag(ami["hvm"], ami["name"])
+                    checkAndAddTag(ami["snapshot"], ami["name"])
+    return
+ 
 def checkAndAddTag(resourceId, region):
-    tagExists = checkTag(resourceId)
+    tagExists = checkTag(resourceId, region)
     if tagExists:
-        print(f"{resourceId} already tagged with FedoraUser=coreos tag")
+        print(f"{resourceId} already tagged with FedoraGroup=coreos tag")
     else:
         addTag(resourceId, region)
-        print(f"FedoraUser=coreos tag successfully added to {resourceId}")    
-
-def checkTag(resourceId):
-    checkTagCmd = f'aws ec2 describe-tags --filters Name=resource-id,Values={resourceId} Name=value,Values=coreos'
+        print(f"FedoraGroup=coreos tag successfully added to {resourceId}")    
+ 
+def checkTag(resourceId, region):
+    checkTagCmd = f'aws ec2 describe-tags --filters Name=resource-id,Values={resourceId} Name=value,Values=coreos Name=key,Values=FedoraGroup --region {region} --output=json'

```

Comment:
Let's try with: `Name=tag:FedoraGroup,Values=coreos`, which is a bit easier to follow.